client/src/app/components/confirmation-overlay/confirmation-overlay.component.ts

```diff
@@ -1,16 +1,48 @@
-import { Component } from '@angular/core';
-import { OverlayComponent } from '../../overlay.service';
+import { Component, OnInit } from '@angular/core';
+import { OverlayComponent, OverlayService } from '../../overlay.service';
 
 @Component({
   selector: 'app-confirmation-overlay',
   templateUrl: './confirmation-overlay.component.html',
   styleUrls: ['./confirmation-overlay.component.scss'],
 })
-export class ConfirmationOverlayComponent implements OverlayComponent {
-  public data: {
-    action: string;
-    actionDescription: string;
-    onConfirmed: () => void;
-    onCancelled: () => void;
+export class ConfirmationOverlayComponent implements OverlayComponent, OnInit {
```

Comment:
I am ware that it's not your code but OverlayComponent is not a good name for a interface definition